src/comments/comments.service.ts

```diff
@@ -1,4 +1,78 @@
-import { Injectable } from '@nestjs/common';
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { CreateCommentDto } from 'src/dto/create-comment.dto';
+import { Post } from 'src/posts/posts.model';
+import { User } from 'src/users/users.model';
+import { Comment } from './comments.model';
 
 @Injectable()
-export class CommentsService {}
+export class CommentsService {
+  constructor(
+    @InjectModel('Posts') private readonly postModel: Model<Post>,
+    @InjectModel('Users') private readonly userModel: Model<User>,
+    @InjectModel('Comments') private readonly commentModel: Model<Comment>,
+  ) {}
+
+  async create(comment: CreateCommentDto): Promise<any> {
+    try {
+      const newComment = new this.commentModel(comment);
+      const createdComment = await newComment.save();
+      return createdComment;
+    } catch (error) {
+      throw new HttpException('Comment not created', HttpStatus.BAD_REQUEST);
+    }
+  }
+
+  async delete(id: string, userId: string): Promise<any> {
+    try {
+      const comment = await this.commentModel.findById(id);
```

Comment:
We don't need to do this, you can just deduce from `findByIdAndDelete` response to deduce if an item was deleted or not.